### ISSUE TITLE
Only add items to updates queue once

### ISSUE THREAD

**Issue by dylwhich** [umbc-hackafe/bridgesim#35]
In the wrapped constructor, API-exposed objects are added to the updates queue twice for some reason, sometimes. This might be because the constructor is being wrapped at two levels in the hierarchy. Actually, that's almost certainly it.

**Comment by dylwhich** [comment]
Was introduced 224f2b0

**Comment by dylwhich** [comment]
Essentially fixed as of 3e17a56, with several caveats (only applicable for API-registered types that have separately-registered subtypes). In order to ensure that an object is only queued for updates once it is **fully** instantiated, every constructor along the type hierarchy between the lowest-level class that has its own API behavior defined (i.e., it has readable, writable, or expose decorators, or has a different Context/autocontext) and the base type (or, if the base type is not registered, then its first registered successor) **must** call super().__init__. Also, all types **must** be registered before their subtypes.

TL;DR: Registered subtypes need to call super().__init__() and types should only be registered after their supertypes are registered (unless the supertype is not exposed, of course). These constraints cannot be enforced, so the API will just subtly break if they are not followed.
